lib/note-content-editor.tsx

```diff
@@ -425,6 +427,42 @@ class NoteContentEditor extends Component<Props> {
     this.editor = editor;
     this.monaco = monaco;
 
+    monaco.languages.registerLinkProvider('plaintext', {
+      provideLinks: (model) => {
+        const matches = model.findMatches(
+          'simplenote://note/[a-zA-Z0-9-]+',
+          true, // searchOnlyEditableRange
+          true, // isRegex
+          false, // matchCase
+          null, // wordSeparators
+          false // captureMatches
+        );
+        return {
+          // don't set a URL on these links, because then Monaco skips resolveLink
+          // @cite: https://github.com/Microsoft/vscode/blob/8f89095aa6097f6e0014f2d459ef37820983ae55/src/vs/editor/contrib/links/getLinks.ts#L43:L65
+          links: matches.map(({ range }) => ({ range })),
+        };
+      },
+      resolveLink: (link) => {
+        const href = editor.getModel()?.getValueInRange(link.range);
+        if (!href) {
+          return;
+        }
+        const match = /^simplenote:\/\/note\/(.+)$/.exec(href);
+        if (!match) {
+          return;
+        }
+
+        const [fullMatch, linkedNoteId] = match;
+
+        if (this.props.notes.has(linkedNoteId as T.EntityId)) {
+          this.props.openNote(linkedNoteId as T.EntityId);
```

Comment:
do we need to do this here or would it make sense to run it by trapping other links? I don't know the answer, I'm just asking to make sure we're not coupling the linking behavior inside the editor with app behaviors outside of the editor. for instance, do we already trap clicked-on links from the editor to intercept them?

Reply:
No, since the simplenote:// links aren't real links I don't think we can trigger the link interceptor from here.

Reply:
That happens over here from the preview component, but this works because they've been transformed into `<a>` tags for preview: https://github.com/Automattic/simplenote-electron/blob/b325afcf2685286d752195a862f6cad4f12ed908/lib/components/note-preview/index.tsx#L84

Reply:
I guess we probably do need that check within the preview component as well, though